File: scheduler.js
const express = require("express");
const cron = require("node-cron");
const bodyParser = require("body-parser");

const app = express();
app.use(bodyParser.json());

let tasks = [];

// Route to add task to the scheduler
app.post("/schedule", (req, res) => {
    const {name, time, repeat} = req.body;

    if (!name || ! time) {
        return res.status(400).json({error: "Task name and time required"});     
    }

    let cronTime = repeat ? time : `*/${time} * * * * *`; // Default to seconds
    const task = cron.schedule(cronTime, () =>{
        console.log(`Executing task${name}`)
    });

    task.push([name, cronTime, task]);
    res.json({message: "Task Scheduled", name, cronTime});
});

// Route to list all the tasls
app.get("/tasks", (req, res) =>{
    res.json(tasks.map(t=>({name: t.name, cronTime: t.cronTime})));
})

// Route to stop a task
app.delete("/stop/:name", (req, res) =>{
    const name = req.params.name;
    const taskIndex = tasks.findIndex(t => t.name === name);

    if (taskIndex === -1){
        return res.status(404).json({error: "Task not Found!"});
    }

    tasks[taskIndex].task.stop();
    tasks.splice(taskIndex, 1);
    res.json({mesage:`Task ${name} stopped and Removed.`});
});

const PORT = 3000;
app.listen(PORT, ()=> console.log(`Task Scheduler running in port ${PORT}`))

File: Python/scheduler.py
from flask import Flask , request, jsonify
from apscheduler.schedulers.background import BackgroundScheduler
import time

app = Flask(__name__)

# Function to excecute the task
def execute_task(name):
    print(f" Executing Task: {name}")

# Route to schedule the task
@app.route("/schedule", methods = ["POST"])
def schedule_task():
    data = request.get_json()
    name = data.get("name")
    interval = data.get("time") # Time in seconds

    if not name or not interval:
        return jsonify({"error": "Task name and interval are required!"}), 400
    
    job = scheduler.add_job(execute_task, 'interval', seconds = int(interval), id=name, args=[name])
    tasks[name] = job
    return jsonify({"message": f"Task {name} scheduled successfully!"}), 200

#Route to list all the tasks
@app.route("/tasks", methods =["GET"])
def list_tasks():
    return jsonify(list(tasks.keys()))

# Route to remove the task
@app.route("stop/<name>", methods = ["DELETE"])
def stop_task(name):
    if name in tasks:
        tasks[name].remove()
        del tasks[name]
        return jsonify({"message": f"Task {name} stopped successfully!"}), 200
    return jsonify({"error": f"Task {name} not found!"}), 404

if __name__ == "__main__":
    app.run(debuhg = True, port = 5000)
